gcs/deleteFile/deleteFile.go

```diff
@@ -0,0 +1,160 @@
+	p.res = make(map[string]*metrics.EventMetrics)
+
+	return nil
+}
+
+// Start starts and runs the probe indefinitely.
+func (p *Probe) Start(ctx context.Context, dataChan chan *metrics.EventMetrics) {
+	probeTicker := time.NewTicker(p.opts.Interval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			probeTicker.Stop()
+			return
+		case <-probeTicker.C:
+			// On probe tick, write data to the channel and run probe.
+			for _, em := range p.res {
+				dataChan <- em
+			}
+
+			// TODO: CHANGE THIS BIT TO OUR PROBE IMPLEMENTATION
+			p.targets = endpoint.NamesFromEndpoints(p.opts.Targets.ListEndpoints())
+			p.initProbeMetrics()
+			probeCtx, cancelFunc := context.WithDeadline(ctx, time.Now().Add(p.opts.Timeout))
+			p.runProbe(probeCtx)
+			cancelFunc()
+		}
+	}
+}
+
+// TODO: Change this!!
+// initProbeMetrics initializes missing probe metrics.
+func (p *Probe) initProbeMetrics() {
+	for _, target := range p.targets {
+		if p.res[target] != nil {
+			continue
+		}
+		var latVal metrics.Value
+		if p.opts.LatencyDist != nil {
+			latVal = p.opts.LatencyDist.Clone()
+		} else {
+			latVal = metrics.NewFloat(0)
+		}
+		p.res[target] = metrics.NewEventMetrics(time.Now()).
+			AddMetric("total", metrics.NewInt(0)).
+			AddMetric("success", metrics.NewInt(0)).
+			AddMetric("latency", latVal).
+			AddLabel("ptype", "redis").
+			AddLabel("probe", p.name).
+			AddLabel("dst", target)
+	}
+}
+
+// TODO: Improve this by passing the storage client variable/making it accessible to reduce num of conns.
+// TODO: CHANGE THIS - THIS IS THE MAIN PROBE RUN() function
+// runProbeForTarget runs probe for a single target.
+func (p *Probe) runProbeForTarget(ctx context.Context, target string) error {
+	bucket := p.c.GetBucket()
+	object := p.c.GetObject()
+
+        client, err := storage.NewClient(ctx)
+        if err != nil {
+                return fmt.Errorf("storage.NewClient: %v", err)
+        }
+        defer client.Close()
+
+        ctx, cancel := context.WithTimeout(ctx, p.opts.Timeout)
+        defer cancel()
+
+        o := client.Bucket(bucket).Object(object)
+        if err := o.Delete(ctx); err != nil {
+                return fmt.Errorf("Object(%q).Delete: %v", object, err)
+        }
+
+	glog.Infof("Object %v deleted in bucket %s.", o, bucket)
+	return nil
+}
+
+// runProbe runs probe for all targets and update EventMetrics.
+func (p *Probe) runProbe(ctx context.Context) {
+	p.targets = endpoint.NamesFromEndpoints(p.opts.Targets.ListEndpoints())
+
+	var wg sync.WaitGroup
+	for _, target := range p.targets {
+		wg.Add(1)
+
+		go func(target string, em *metrics.EventMetrics) {
+			defer wg.Done()
+			em.Metric("total").AddInt64(1)
+			start := time.Now()
+			err := p.runProbeForTarget(ctx, target) // run probe just for a single target
+			if err != nil {
+				p.l.Errorf(err.Error())
+				return
+			}
+			em.Metric("success").AddInt64(1)
+			em.Metric("latency").AddFloat64(time.Now().Sub(start).Seconds() / p.opts.LatencyUnit.Seconds())
```

Comment:
I would hope that GCS is fast enough that latency should be in milliseconds :-P

Reply:
I have removed the deleteFile.go from this PR, as I want to merge it in a different PR.